rust/connlib/tunnel/src/device_channel/tun_android.rs

```diff
@@ -175,3 +166,17 @@ impl Closeable {
         self.closed.store(true, Ordering::Release);
     }
 }
+
+fn ipv4(ip: &IpNetwork) -> Option<&Ipv4Network> {
+    match ip {
+        IpNetwork::V4(v4) => Some(v4),
+        IpNetwork::V6(_) => None,
+    }
+}
+
+fn ipv6(ip: &IpNetwork) -> Option<&Ipv6Network> {
+    match ip {
+        IpNetwork::V4(_) => None,
+        IpNetwork::V6(v6) => Some(v6),
+    }
+}
```

Comment:
Can we move that up one module so we don't have to duplicate it? Child modules can access private functions in `super`.